contracts/DecentHats.sol

```diff
@@ -0,0 +1,481 @@
+     *
+     * @dev In order for the module to be able to create hats on behalf of the Safe, the Safe must first
+     * transfer its top hat to this contract. This function transfers the top hat back to the Safe after
+     * creating the role hat.
+     *
+     * @dev The function simply calls `_createTermedHatAndAccountAndMintAndStreams` and then transfers the top hat back to the Safe.
+     *
+     * @dev Stream funds on Termed Roles are targeted directly at the nominated wearer.
+     * The wearer should directly call `withdraw-` on the Sablier contract.
+     *
+     * @dev Termed Role hat creation, minting, and stream creation are handled here in order
+     * to avoid a race condition where not more than one active proposal to create a new termed role can exist at a time.
+     * See: https://github.com/decentdao/decent-interface/issues/2402
+     */
+    function createTermedRoleHat(
+        CreateTermedRoleHatParams calldata params
+    ) external {
+        _createTermedHatAndAccountAndMintAndStreams(
+            params.hatsProtocol,
+            params.topHatAccount,
+            _createElectionEligiblityModule(
+                params.hatsModuleFactory,
+                params.hatsElectionEligibilityImplementation,
+                params.hatsProtocol.getNextId(params.adminHatId),
+                params.topHatId,
+                params.hat.termedParam
+            ),
+            params.adminHatId,
+            params.hat
+        );
+
+        params.hatsProtocol.transferHat(
+            params.topHatId,
+            address(this),
+            msg.sender
+        );
+    }
+
+    /* /////////////////////////////////////////////////////////////////////////////
+                        INTERAL FUNCTIONS
+    ///////////////////////////////////////////////////////////////////////////// */
+
+    function _declareSafeHatTree(
+        address _keyValuePairs,
+        uint256 topHatId
+    ) internal {
+        string[] memory keys = new string[](1);
+        string[] memory values = new string[](1);
+        keys[0] = "topHatId";
+        values[0] = Strings.toString(topHatId);
+
+        IAvatar(msg.sender).execTransactionFromModule(
+            _keyValuePairs,
+            0,
+            abi.encodeWithSignature(
+                "updateValues(string[],string[])",
+                keys,
+                values
+            ),
+            Enum.Operation.Call
+        );
+    }
+
+    function _createHat(
+        IHats _hatsProtocol,
+        uint256 adminHatId,
+        Hat memory _hat,
+        address toggle,
+        address eligibility
+    ) internal returns (uint256) {
+        return
+            _hatsProtocol.createHat(
+                adminHatId,
+                _hat.details,
+                _hat.maxSupply,
+                eligibility,
+                toggle,
+                _hat.isMutable,
+                _hat.imageURI
+            );
+    }
```

Comment:
Is it benefitting us to have this single external function call wrapped in an internal function? Why not just inline the `_hatsProtocol.crateHat()` call wherever it's used?

Reply:
fixed.